### DISCUSSION TITLE
Gauge + Java + Allure Reporting

### DISCUSSION THREAD

**Discussion by omutwar** [getgauge/taiko#2689] [General]
Hello Fellow Coders:

Currently one of the projects I am working on wanted to add Allure reporting to Gauge framework with Java as programming language. 

I was wondering if anyone who has been working with Gauge, happened to have tested with Allure, or have some idea how I may be able to integrate Gauge with Allure reporting? 

Please share your thoughts and ideas. That would be deeply appreciated!!

**Comment by AdrianVuncannonD** [comment]
Yes, it is possible to integrate Allure reporting with Gauge framework in Java.

**Reply by omutwar** [reply]
Hello Adrian, 

is it possible to share the specific steps to set up the Allure reporting for Gauge+Java framework?

what dependencies we need to add, how to configure the POM.xml ... etc

Thank you

**Reply by slymanAktas** [reply]
You don't need to configure nothing. 
- First you need to set up allure app in your pc.(Download binary then path it to System Environment)
- Gauge create xml report under 'reports/xml-report' after execution.
- So you can create allure result with this xml file via this command -> allure generate path/xml-report --clean -o allure-report
- Unfortunately this report file doesn't include failed test case screen shot
- Because there is no information about screen shot path in result.xml
- https://github.com/getgauge/gauge/issues/2438

**Reply by omutwar** [reply]
Thank you Slyman!

Although has been a while, still working on this on the side.

If you or others happen to know a better source about Gauge besides their official site, let me know. It seems like their site starts with Java and wanders off to JavaScript. As of this framework is not maintained at all.

**Comment by erdemkiiliic** [comment]
Yes, with the xml report generated by Gauge, a new report can be created using the Allure plugin. But I don't know how to do the same for html report.

**Reply by Hungleminh** [reply]
Hi @erdemkiiliic , do you have anything news for this issue with html report?
And how about xml-report with Gauge? In my case, it doesn't provide much information such as failed, screenshot after generating to allure report.

Thanks,
